specutils/fitting/fitmodels.py

```diff
@@ -382,12 +384,14 @@ def _fit_lines(spectrum, model, fitter=fitting.LevMarLSQFitter(calc_uncertaintie
 
     if isinstance(weights, str):
         if weights == 'unc':
-            uncerts = spectrum.uncertainty
-
-            # Astropy fitters expect weights in 1/sigma
-            if uncerts is not None:
+            if spectrum.uncertainty is not None:
+                # Convert uncertainty to StdDev, then invert
+                uncerts = StdDevUncertainty(uncertainty._convert_uncertainty(spectrum.uncertainty,
+                                                                             StdDevUncertainty))
+                # Astropy fitters expect weights in 1/sigma
                 weights = uncerts.array ** -1
             else:
+                weights = None
                 warnings.warn("Uncertainty values are not defined, but are "
                               "trying to be used in model fitting.")
```

Comment:
Not something you changed, but this warning could be more informative. Maybe something like "Fitting is set to use uncertainties as weights, but the input spectrum's uncertainty is None"